src/caf/distribute/gravity_model/multi_area.py

```diff
@@ -243,35 +372,67 @@ def _create_seed_matrix(self, cost_distributions, cost_args, params_len):
     # pylint: disable=too-many-locals
     def _calibrate(
         self,
+        distributions: MultiCostDistribution,
+        running_log_path: Path,
+        furness_jac: bool = False,
         diff_step: float = 1e-8,
         ftol: float = 1e-4,
         xtol: float = 1e-4,
+        furness_tol=DEFAULT_FURNESS_TOL,
         grav_max_iters: int = 100,
         failure_tol: float = 0,
         default_retry: bool = True,
         verbose: int = 0,
         **kwargs,
-    ) -> dict[str, GravityModelCalibrateResults]:
-        params_len = len(self.dists[0].function_params)
+    ) -> dict[str | int, GravityModelCalibrateResults]:
+        params_len = len(distributions[0].function_params)
         ordered_init_params = []
-        for dist in self.dists:
+        # TODO validate cost distributions
+        for dist in distributions:
             params = self._order_cost_params(dist.function_params)
             for val in params:
                 ordered_init_params.append(val)
 
+            max_binning = dist.cost_distribution.max_vals.max()
+            min_binning = dist.cost_distribution.min_vals.min()
+
+            max_cost = self.cost_matrix[dist.zones, :].max()
+            min_cost = self.cost_matrix[dist.zones, :].min()
+
+            if max_cost > max_binning:
+                LOG.warning(
+                    f"the maximum cost in the cost matrix for"
+                    f" category {dist.name}, was {max_cost}, "
+                    "whereas the highest bin edge in cost"
+                    f" distribution was {max_binning}, "
+                    "you will not be fitting to trips"
+                    " with a cost greater than the binning"
+                )
+            if min_cost < min_binning:
+                LOG.warning(
+                    f"the min cost in the cost matrix for"
+                    f" category {dist.name}, was {min_cost}, "
+                    "whereas the lowest bin edge in cost"
+                    f" distribution was {min_binning}, "
+                    "you will not be fitting to trips"
+                    " with a cost less than the binning"
+                )
```

Comment:
Use the warnings module instead of logging because it gives the user more control over the warnings, they are captured by the logging from toolkit by default anyway so will still show up in any log files.

In future we might want to define custom GM warnings to make it even easier for the user to handle them in specific ways if needed.
```suggestion
            if max_cost > max_binning:
                warnings.warn(
                    f"the maximum cost in the cost matrix for"
                    f" category {dist.name}, was {max_cost}, "
                    "whereas the highest bin edge in cost"
                    f" distribution was {max_binning}, "
                    "you will not be fitting to trips"
                    " with a cost greater than the binning"
                )
            if min_cost < min_binning:
                warnings.warn(
                    f"the min cost in the cost matrix for"
                    f" category {dist.name}, was {min_cost}, "
                    "whereas the lowest bin edge in cost"
                    f" distribution was {min_binning}, "
                    "you will not be fitting to trips"
                    " with a cost less than the binning"
                )
```